Bottom: 5e869f73ebcb97bc15c4a3f5fcea98b1ddb794af
Top:    6dea2e01f138ccae678f318e20bcacd40e056b19
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-05 18:44:15 +0100

Add the drc flag as a parameter

Ok, this is nasty.. I'm obscuring setting of the global varaible... be careful!


---

diff --git a/src/find.c b/src/find.c
index aa7aabe..3747233 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3324,9 +3324,10 @@ struct drc_info
 static void
 start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
                       int flag, bool AndDraw,
-                      Coord bloat)
+                      Coord bloat, bool is_drc)
 {
   Bloat = bloat;
+  drc = is_drc;
   ListStart (type, ptr1, ptr2, ptr3, flag);
   DoIt (flag, true, AndDraw);
   DumpList ();
@@ -3348,7 +3349,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
 
   if (PCB->Shrink != 0)
     {
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink, false);
       /* ok now the shrunk net has the SELECTEDFLAG set */
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
@@ -3359,10 +3360,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           /* make the flag changes undoable */
           ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
           User = true;
-          drc = false;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink);
-          drc = true;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink, false);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0, true);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3392,9 +3391,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       DumpList ();
     }
   /* now check the bloated condition */
-  drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
-  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0);
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0, false);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3405,10 +3403,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       /* make the flag changes undoable */
       ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
       User = true;
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat, true);
+      User = false;
       drc = false;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0);
-      drc = true;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3427,15 +3425,13 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       pcb_drc_violation_free (violation);
       free (object_id_list);
       free (object_type_list);
-      User = false;
-      drc = false;
       if (!throw_drc_dialog())
         return (true);
       IncrementUndoSerialNumber ();
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
-      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0);
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0, false);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
